### PR TITLE
Fix/tx timeouts

### PR DESCRIPTION
**Pull request by mjkoo**

### CODE REVIEW

File: src/polyswarmd/eth.py [hunk 1]
```diff
@@ -21,10 +21,10 @@
 
 misc = Blueprint('misc', __name__)
 
-# TODO: not sure if this should be hardcoded/fixed; min gas needed for POST to settle bounty
-gas_limit = 5000000
-zero_address = '0x0000000000000000000000000000000000000000'
-transfer_signature_hash = 'a9059cbb'
+MAX_GAS_LIMIT = 500000
+GAS_MULTIPLIER = 1.2
```
Comment (ceseale) [inline src/polyswarmd/eth.py]: Is this supposed to be used somewhere?

File: src/polyswarmd/eth.py [hunk 2]
```diff
@@ -169,11 +170,20 @@ def build_transaction(call, nonce):
...
-        'gas': gas_limit,
+        'gas': MAX_GAS_LIMIT,
     }
+
     if g.chain.free:
         options["gasPrice"] = 0
 
+    try:
+        gas = int(call.estimateGas({'from': g.eth_address, **options}))
+        options['gas'] = min(MAX_GAS_LIMIT, gas)
```
Comment (ceseale) [inline src/polyswarmd/eth.py]: Kinda worried about nonce desyncing issues if the transaction gas is less than `MAX_GAS_LIMIT` and etimateGas produces a gas that's higher the previous with the same nonce. Would that be an issue for us? Might need to see this in action to really know. 

I do know the `etimateGas` can produce somewhat wonky results based on the contract state.
https://ethereum.stackexchange.com/questions/266/what-are-the-limitations-to-estimategas-and-when-would-its-estimate-be-considera

File: src/polyswarmd/eth.py (hunk 1 above)
Comment (mjkoo) [reply src/polyswarmd/eth.py]: Yep good catch

File: src/polyswarmd/eth.py (hunk 2 above)
Comment (mjkoo) [reply src/polyswarmd/eth.py]: gas limit doesnt affect replacement txs only gas price afiak

usually the value will be close unless it fails in which case we fall back to max gas limit, we're supposed to addd 20% or so to be safe but left out that mult

Comment (ceseale) [reply src/polyswarmd/eth.py]: Okay sounds good

**Comment by mjkoo** [comment]
Ref #167 to do this properly

**Pull request merged**